### PR TITLE
Enable linters in golansci-lint and fix issues

### PR DESCRIPTION
**Pull request by Danielius1922**
Enabled linters:
    - nakedret
    - nilerr

### CODE REVIEW

**Review by jkralik — APPROVED** [review]

File: http-gateway/serverMux/jsonMarshaler.go [hunk 1]
```diff
@@ -135,13 +135,13 @@ func modify(v interface{}) (newValue interface{}, wantDelete bool) {
...
 	if err != nil {
-		return data, err
+		return nil, err
 	}
 
 	var val interface{}
 	err = json.Decode(data, &val)
 	if err != nil {
-		return data, nil
+		return nil, err
```
Comment (jkralik) [inline http-gateway/serverMux/jsonMarshaler.go]: `return data, nil`

File: pkg/log/log.go [hunk 2]
```diff
@@ -369,7 +369,7 @@ func (l *WrapSuggarLogger) Fatalf(template string, args ...interface{}) {
 
 func (l *WrapSuggarLogger) LogAndReturnError(err error) error {
 	if err == nil {
-		return err
+		return nil
```
Comment (jkralik) [inline pkg/log/log.go:372]: return `data, nil`

**Review by jkralik — CHANGES_REQUESTED** [review]
look to comments

File: http-gateway/serverMux/jsonMarshaler.go (hunk 1 above)
Comment (Danielius1922) [reply http-gateway/serverMux/jsonMarshaler.go]: Only this line? What about line 157 I think we talked yesterday about that as well.

Comment (Danielius1922) [reply http-gateway/serverMux/jsonMarshaler.go]: Also, I'd like to add comment to the code. But ...I forgot it. :D Could you please formulate a short comment about why we return data in this case?

**Review by jkralik — APPROVED** [review]

**Pull request merged**